apollo-runtime/src/main/java/com/apollographql/android/impl/ResponseBodyConverter.java

```diff
@@ -53,6 +59,7 @@
         }
       }
       jsonReader.endObject();
+      this.cache.cacheStore().merge(responseNormalizer.records());
```

Comment:
Resurfacing some comments from earlier: 

I still feel like the right thing is to either store all, or none of the records from a response, and not burden cachestore implementations with the concept of transactions, or rollbacks.

I also don't love that the ResponseBodyConverter is saving the records at all. I feel like it's job should be to create the records as the response is parsed, and then return them with the Data.